### ISSUE TITLE
kmdclient.SignTransaction can't accept Transaction object

### ISSUE THREAD

**Issue by hernandp** [algorand/js-algorand-sdk#284]
My use-case is to use a KMD wallet, take the first key (0) as sender and create an application call TX.  The problem is that the call:

```
    const signedTx= await kmdclient.signTransaction(cfg.walletHandle, cfg.walletPwd, tx);
```
will throw **MALFORMED ADDRESS** and inspecting it a bit I found that he expects a `string` type address but I'm getting which is I guess an already decoded address: 

```
tx.from
{publicKey: Uint8Array(32), checksum: Uint8Array(4)}
```

From my analysis, the following call will already construct a `Transaction` object with a `from` field containing a decoded address, via  using the `Transaction`constructor:

```
  const txn = algosdk.makeApplicationCreateTxn(sender, params, algosdk.OnApplicationComplete.NoOpOC,
    approvalBytecode,
    clearStateBytecode,
    creationData.localInts,
    creationData.localByteslices,
    creationData.globalInts,
    creationData.globalByteslices);
  const txId = txn.txID().toString()
```
Before calling this, sender is `73TOYILA2RT3QGFMBLL256HAL3XYSJMBDHXRLOVDOB7WX3QLFUT4TOESKA`.
After the call, txn will have the from field set as decoded.

So when kmdClient.signTransaction is called, it will try to build an already decoded `from field`   (kmd.js:255):

```
this.signTransaction = async function (walletHandle, walletPassword, transaction) {
        let tx = new txn.Transaction(transaction);
```

It seems that this will happen in the same way  using any of the `makeTxn...` functions before the `kmdclient.SignTransaction` call.

Thanks in advance.

**Comment by hernandp** [comment]
I'm using 1.8.1.

Another side-effect of the kmd.js:255 `new txn.Transaction(transaction)` call is due to this piece in `transaction.js:74`:

```
if (appArgs !== undefined) {
            if (!Array.isArray(appArgs)) throw Error("appArgs must be an Array of Uint8Array.");
            appArgs.forEach((arg) => {
                if (arg.constructor !== Uint8Array) throw Error("each element of AppArgs must be a Uint8Array.");
            });
        } else {
            appArgs = new Uint8Array(0);
        }
```

It will throw `appArgs must be an Array of Uint8Array.` because the `appArgs` var created in the else stmt is `Uint8Array(0)[]` and that is not evaluated as true by `Array.isArray( ... )`.

Thanks.

**Comment by jasonpaulos** [comment]
@hernandp thanks for reporting this! It looks like the `signTransaction` method of the KMD client is not able to accept an instance of the `Transaction` classes yet. We should fix this so that it's more usable.

**Comment by jdtzmn** [comment]
Closed by #318.
